pkg/providers/v1/aws.go

```diff
@@ -4371,8 +4375,24 @@ func (c *Cloud) GetLoadBalancer(ctx context.Context, clusterName string, service
 
 // GetLoadBalancerName is an implementation of LoadBalancer.GetLoadBalancerName
 func (c *Cloud) GetLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) string {
-	// TODO: replace DefaultLoadBalancerName to generate more meaningful loadbalancer names.
-	return cloudprovider.DefaultLoadBalancerName(service)
+	if val := service.Annotations[ServiceAnnotationLoadBalancerName]; val != nil {
+		r := regexp.MustCompile(`^[a-zA-Z0-9-]{0,32}`)
+		if r.MatchString(val) && val[0:1] != "-" && val[len(val)-1:] != "-" {
+			return val
+		} else {
+			klog.Error("Error : AWS load balancer name, a maximum of 32 alphanumeric characters including hyphens are allowed, but the name must not begin or end with an hyphen.")
+		}
+		return val
+	}
+
+	//AWS requires that the name of a load balancer is shorter than 32 bytes.
+	//Default implementation, currently deprecated
+	ret := "a" + string(service.UID)
+	ret = strings.Replace(ret, "-", "", -1)
+	if len(ret) > 32 {
+		ret = ret[:32]
+	}
+	return ret
```

Comment:
Instead of the above code, simply fallback to:

```go
return cloudprovider.DefaultLoadBalancerName(service)
```

Reply:
Hello @jaypipes, for this one, I was wondering what to do exactly, on the parent project, the method is marked as deprecated, so should we really use the method, or implement it instead here ? 
https://github.com/kubernetes/cloud-provider/blob/c819ca6bd71c22c47e38c26461a0a0db96d8b9b0/cloud.go#L84